test/gtest/coll/test_gatherv.cc

```diff
@@ -0,0 +1,295 @@
+    const gtest_ucc_inplace_t inplace  = std::get<5>(GetParam());
+    UccTeam_h                 team     = UccJob::getStaticTeams()[team_id];
+    int                       size     = team->procs.size();
+    UccCollCtxVec             ctxs;
+
+    set_inplace(inplace);
+    SET_MEM_TYPE(mem_type);
+    set_root(root);
+
+    data_init(size, dtype, count, ctxs, false);
+    UccReq req(team, ctxs);
+    req.start();
+    req.wait();
+    EXPECT_EQ(true, data_validate(ctxs));
+    data_fini(ctxs);
+}
+
+UCC_TEST_P(test_gatherv_0, single_persistent)
+{
+    const int                 team_id  = std::get<0>(GetParam());
+    const ucc_datatype_t      dtype    = std::get<1>(GetParam());
+    const ucc_memory_type_t   mem_type = std::get<2>(GetParam());
+    const int                 count    = std::get<3>(GetParam());
+    const int                 root     = std::get<4>(GetParam());
+    const gtest_ucc_inplace_t inplace  = std::get<5>(GetParam());
+    UccTeam_h                 team     = UccJob::getStaticTeams()[team_id];
+    int                       size     = team->procs.size();
+    const int                 n_calls  = 3;
+    UccCollCtxVec             ctxs;
+
+    set_inplace(inplace);
+    SET_MEM_TYPE(mem_type);
+    set_root(root);
+
+    data_init(size, dtype, count, ctxs, true);
+    UccReq req(team, ctxs);
+
+    for (auto i = 0; i < n_calls; i++) {
+        req.start();
+        req.wait();
+        EXPECT_EQ(true, data_validate(ctxs));
+        reset(ctxs);
+    }
+
+    data_fini(ctxs);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    , test_gatherv_0,
+    ::testing::Combine(::testing::Range(1, UccJob::nStaticTeams), // team_ids
+                       PREDEFINED_DTYPES,
+#ifdef HAVE_CUDA
+                       ::testing::Values(UCC_MEMORY_TYPE_HOST,
+                                         UCC_MEMORY_TYPE_CUDA),
+#else
+                       ::testing::Values(UCC_MEMORY_TYPE_HOST),
+#endif
+                       ::testing::Values(1, 3, 8192), // count
+                       ::testing::Values(0, 1),       // root
+                       ::testing::Values(TEST_INPLACE, TEST_NO_INPLACE)));
+
+class test_gatherv_1 : public test_gatherv,
+                      public ::testing::WithParamInterface<Param_1> {
+};
+
+UCC_TEST_P(test_gatherv_1, multiple_host)
+{
+    const ucc_datatype_t       dtype    = std::get<0>(GetParam());
+    const ucc_memory_type_t    mem_type = std::get<1>(GetParam());
+    const int                  count    = std::get<2>(GetParam());
+    const int                  root     = std::get<3>(GetParam());
+    const gtest_ucc_inplace_t  inplace  = std::get<4>(GetParam());
+    std::vector<UccReq>        reqs;
+    std::vector<UccCollCtxVec> ctxs;
+
+    for (int tid = 0; tid < UccJob::nStaticTeams; tid++) {
+        UccTeam_h     team = UccJob::getStaticTeams()[tid];
+        int           size = team->procs.size();
+        UccCollCtxVec ctx;
+
+        if (size == 1 && root > 0) {
```

Comment:
replace the predicate by
`size < root`
same in scatterv tests

This check should also be done in the other tests, right?

Reply:
Replaced with size <= root (also when size == root its invalid). Added in all gtests for gatherv and scatterv. Will add for all other collective gtests in seperate PR.